packages/strapi-generate-api/templates/bookshelf/service.template

```diff
@@ -56,8 +58,10 @@ module.exports = {
    * @return {Promise}
    */
 
-  add: (values) => {
-    return <%= globalID %>.forge(values).save();
+  add: async (values) => {
+    const data = await <%= globalID %>.forge(_.omit(values, _.keys(_.groupBy(strapi.models.<%= id %>.associations, 'alias')))).save();
+    await strapi.hook.bookshelf.load().manageRelations(strapi.models, <%= globalID %>, _.merge(_.clone(data.toJSON()), { values }));
```

Comment:
You don't need to send `strapi.models` because it's available globally. Not a big fan to send the model available globally through the function's parameters. You can only send the model name. Also, we need to simplify the syntax to call the `manageRelations` method.

```js
// Use the query method and make the method ORM agnostic.
await strapi.query('<%= globalID %>').manageRelations(_.merge(_.clone(data.toJSON()), { values }));

// Make the manageRelations a first class and ORM agnostic method.
await strapi.manageRelations('<%= globalID %>').manageRelations(_.merge(_.clone(data.toJSON()), { values }));

// Remove the load() call by making a shortcut.
await strapi.hook.bookshelf.manageRelations('<%= globalID %>', _.merge(_.clone(data.toJSON()), { values }));
```